src/classes/RD2_ScheduleService.cls

```diff
@@ -125,26 +125,45 @@ public without sharing class RD2_ScheduleService {
         return rd.npe03__Installment_Period__c == RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH;
     }
 
+
     /***
     * @description Synchronizes the schedule records to match an updated recurring donation
+    * and returns only schedules that require to be updated or created.
     * @param rd Recurring Donation record
     * @param schedules Recurring Donation Schedule record
     * @return TDTM_Runnable.DMLWrapper
     */
     public TDTM_Runnable.DMLWrapper getUpdatedSchedules(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> schedules) {
+        Boolean isUpdate = true;
+        return rebuildSchedules(rd, schedules, isUpdate);
+    }
 
+    /***
+    * @description Synchronizes the schedule records to match an updated recurring donation
+    * @param rd Recurring Donation record
+    * @param schedules Recurring Donation Schedule record
+    * @param isUpdate Indicates if returned schedules should be only changed and new schedules
+    * @return TDTM_Runnable.DMLWrapper
+    */
+    private TDTM_Runnable.DMLWrapper rebuildSchedules(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> schedules, Boolean isUpdate) {
```

Comment:
@npsp-reedestockton , the method is renamed to return all active schedules including pause schedule to the trigger in before context. Only changed/new schedules are returned when the actual schedule happens in the RD update after context. Let me know if there is any issue with this.